packages/model-viewer/src/test/features/animation-spec.ts

```diff
@@ -103,6 +103,22 @@ suite('ModelViewerElementBase with AnimationMixin', () => {
           element.currentTime = 5;
           expect(element[$scene].shouldRender()).to.be.true;
         });
+
+        suite('when play is invoked again with options', () => {
+          setup(async () => {
+            const animationsPlay = waitForEvent(element, 'play');
+            element.play({repitations: 10, pingpong: true});
```

Comment:
Good start, but this still isn't really testing that what you've  asked for has occurred. I would recommend you serve our animations page locally, select the element in dev tools and try interacting with it by calling `play` and `pause` and try your different arguments. I still have the same problems: calling `play()` with arguments when it's already playing has no effect (this test passes because the animation continues  to play and you never check if it stops). If I pause and `play({repetitions: 1})`, it does stop correctly, but can never be restarted with `play`.

Reply:
Thanks for the advice. I serve animations page locally and confirmed that it works.
However, I don't know how to write the test code. Is there any way to know when the animation has completed the specified number of times? Or should I add such an event? (for example `finished`)

Reply:
I would modify this test to set repetitions to 1, then set some timeouts: after 80% of the animation duration, check that `animationIsPlaying`, after 180% duration, check that it's still playing and the current time is less than last time (playing backwards). After 220%, `animationIsPlaying` should be false and time back to zero. Maybe another version with pingpong false. Choose a nice short  animation (maybe "Punch" from the Robot) so  the test completes quickly.

Reply:
Added a test to call `play({repetitions: 2, pingpong: true})` with `animationName` set to "Punch".
It verifies that the state at 80%, 180%, and 220% elapsed time is the intended one.